client/components/SingleProduct.js

```diff
@@ -0,0 +1,44 @@
+import React from 'react';
+import { connect } from 'react-redux';
+import { Component } from 'react';
+import { fetchProduct } from '../store/products'
+
+
+class SingleProduct extends Component {
+    async componentDidMount () {
+         //console.log('THE PROPS', this.props)
+         const productId = this.props.match.params.id
+         await this.props.getProduct(productId);
+        
+    }
+    render () {
+        const { selectedProduct } = this.props.products;
+        //  console.log(selectedProduct);
```

Comment:
Let me try out the code first and understand how this works. I guess when we click on the product tile in AllProducts view we should be assigning that product to 'selectedProduct' ? Is that what's happening here? I'm just confused what 'selectedProduct' refers to in this configuration

Reply:
Yeah, check out the fetchProduct thunk and the setSelectedProduct action creator in `client/store/products.js`